### DISCUSSION TITLE
How to sign a Mac application with a Developer ID certificate?

### DISCUSSION THREAD

**Discussion by onaxon** [wailsapp/wails#721] [Q&A] [answered]
I would like to sign a Mac application with a Developer ID certificate, the information I have found is for applications developed with XCode, or Electron applications for which there are utilities. Can someone tell me the steps to sign a version 1 Wails application or give me any clues?

**Comment by leaanthony** [comment, marked as answer]
It would be the same way as signing any other Go application. I've seen success with [Gon](https://github.com/mitchellh/gon)

**Comment by onaxon** [comment]
After several attempts it seems to work, but I get a little out of the gon instructions.

First I had to generate an app-specific password following [these instructions](https://www.imore.com/how-generate-app-specific-passwords-iphone-ipad-mac).
 
The password generated is the one that I have included in the configuration file, not the password corresponding to the apple id.

My sign.json config file:
```json
{
	"source": [
		"./eBooky.app"
	],
	"bundle_id": "es.axon.ebooky",
	"apple_id": {
		"username": "{my apple id email]",
		"password": "{the app generated password}"
	},
	"sign": {
		"application_identity": "Developer ID Application: Pedro Marban"
	},
	"dmg": {
		"output_path": "ebooky v1.2.dmg",
		"volume_name": "eBooky"
	}
}
```

The console output:
![image](https://user-images.githubusercontent.com/913775/119550504-c9029380-bd98-11eb-9dcd-e300fd0ac990.png)

I omitted the zip because zips can't be stapled.

I said it seemed to work because I have not managed to install it on older versions of MacOS, but that is subject for another question.

**Reply by leaanthony** [reply]
I feel like at this point it might be worth asking for help from the Gon developers. I wish I could help more!
